pkg/metrics/timer.go

```diff
@@ -0,0 +1,60 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	startTime    time.Time
+	totalTime    time.Time
+	listOfStages []string
+	stageTimes   map[string]time.Duration
+}
+
+const (
+	// InfrastructureStageName is a constant string value that is used to display the Infrastructure created stage text in the logs.
+	InfrastructureStageName = "Infrastructure"
+	// APIStageName is a constant string value that is used to display the API stage text in the logs.
+	APIStageName = "API"
+	// BootstrapCompleteStageName is a constant string value that is used to display the Bootstrap Complete text in the logs.
+	BootstrapCompleteStageName = "Bootstrap Complete"
+	// BootstrapDestroyStageName is a constant string value that is used to display the Boostrap Destroy Complete text in the logs.
+	BootstrapDestroyStageName = "Bootstrap Destroy"
+	// ClusterOperatorStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ClusterOperatorStageName = "Cluster Operator"
+
+	// ConsoleStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ConsoleStageName = "Console"
+)
+
+var timer Timer
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer() {
+	timer = Timer{}
+	timer.startTime = time.Now().Round(time.Second)
+	timer.totalTime = time.Now().Round(time.Second)
+	timer.listOfStages = []string{}
+	timer.stageTimes = make(map[string]time.Duration)
+}
+
+// SetTimeElapsed returns the duration of the current stage.
+func SetTimeElapsed(key string) {
+	duration := time.Now().Round(time.Second).Sub(timer.startTime)
+	timer.listOfStages = append(timer.listOfStages, key)
+	timer.stageTimes[key] = duration
+	timer.startTime = time.Now().Round(time.Second)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	logrus.Debugf("Time elapsed for each section")
```

Comment:
This should not be printed when destroy or any other command with no stages runs.

Reply:
Well this function will only have to be called during the exit of the command that we want to log.

Reply:
Currently, if you run `openshift-install destroy cluster` the output is:

```
Time elapsed for each section
Total time elapsed : 4m29s"
```
"Time elapsed for each section" is confusing here and should not be printed when destory or another command with no stages is run.

There should be a colon at the end. Also consider playing around with the wording, perhaps just `Time elapsed:` It is probably better to avoid the word "section" in an official capacity.